lib/geocoder/lookups/esri.rb

```diff
@@ -52,7 +52,14 @@ def token
         configuration[:token].to_s
       elsif configuration.api_key # generate a new token if we have credentials
         token_instance = Geocoder::EsriToken.generate_token(*configuration.api_key)
-        Geocoder.configure(:esri => {:token => token_instance})
+
+        esri_config = {:token => token_instance}
+        if configuration[:for_storage]
+          esri_config[:for_storage] = configuration[:for_storage]
+        end
+        # replaces the entire :esri config section with the new object
+        Geocoder.configure(:esri => esri_config)
```

Comment:
The existing code was already overwriting the whole config, which is what was causing the problem. From what I can tell, there aren't any other options that can be provided in the config other than `token` and `for_storage`.